sys-devel/gcc_bootstrap/gcc_bootstrap-11.2.0_2021_07_28.recipe

```diff
@@ -232,48 +234,61 @@ BUILD()
 		"../config.h"
 	sed -i -e 's,#define _GLIBCXX_HAVE_TLS 1,/* #undef _GLIBCXX_HAVE_TLS */,' \
 		"../include/$effectiveTargetMachineTriple/bits/c++config.h"
-	make CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
+	make $jobArgs CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
 	mv .libs/libsupc++.a .libs/libsupc++-kernel.a
 	mv saved/libsupc++* .libs/
 	mv saved/gthr-default.h "../include/$effectiveTargetMachineTriple/bits/"
 	mv saved/config.h ..
 	mv saved/c++config.h "../include/$effectiveTargetMachineTriple/bits/"
 	rmdir saved
 
+
 	## BOOTLOADER libgcc + libsupc++
 	local bootCcFlags
 	if [ $effectiveTargetArchitecture == arm ]; then
 		# EFI arm (32-bit) requires software fp
-		bootCcFlags+="-mfloat-abi=soft";
+		bootCcFlags+="-mfloat-abi=soft -nostartfiles -fshort-wchar";
 	fi
 
+	echo "### libgcc-boot ###"
+
 	# build bootloader version of libgcc and libgcc_eh.a (no threads, TLS)
 	cd $objectsDir/$effectiveTargetMachineTriple/libgcc
 	mkdir -p saved
 	mv libgcc.a libgcc_eh.a libgcc_s* libgcov* libgcc-kernel.a libgcc_eh-kernel.a saved/
 	make clean
 	ln -sfn "$sourceDir/libgcc/gthr-single.h" gthr-default.h
-	make CFLAGS="-O2 $bootCcFlags" CXXFLAGS="-O2 $bootCcFlags"
+	make $jobArgs CFLAGS="-O2 $bootCcFlags" CXXFLAGS="-O2 $bootCcFlags"
 	mv libgcc.a libgcc-boot.a
 	mv libgcc_eh.a libgcc_eh-boot.a
+	ln -sfn "$sourceDir/libgcc/gthr-posix.h" gthr-default.h
 	mv saved/* .
 	rmdir saved
 
+	echo "### libsupc++-boot ###"
+
 	# build bootloader version of libsupc++.a (without threads or TLS)
 	cd $objectsDir/$effectiveTargetMachineTriple/libstdc++-v3/libsupc++
 	mkdir -p saved
 	mv .libs/libsupc++* saved/
+	cp "../include/$effectiveTargetMachineTriple/bits/gthr-default.h" \
```

Comment:
@X547  indicated the presence of gthr was an indicator of a non-pthread supporting toolchain.   This feels related?

Reply:
Not quite sure about this one. Tbh I just copied it from previous version. 
There's gthr-posix.h, gthr-single.h and gthr-default.h with the last one being identical to one of the first two.
So this code just swaps it around and then after building the library it swaps it back to the original state.